### ISSUE TITLE
Improve external auth documentation

### ISSUE THREAD

**Issue by ChristopherJTrent** [coder/coder#15968]
At present, the [external auth docs](https://coder.com/docs/admin/external-auth) are great for the provider-side setup (E.G. creating a github oauth app.) However, they're rather severely lacking in information regarding the coder-side setup. There is currently no mention of what environment needs to have the given environment variables set (Template level/coder server process level). It's also unclear how one would go from having an external auth provider set up to being able to use it in a template, especially if you're using dev containers that are part of private github repositories (though I acknowledge that *that* discussion is perhaps better had in the envbuilder repo.)

In short, would it be possible to either have an update made to the external auth docs, or for a more in-depth guide/quickstart to be made on the matter?

**Comment by matifali** [comment]
cc: @EdwardAngert

**Comment by ChristopherJTrent** [comment]
Thanks to everyone from the coder team who took time out of their day to work on this. It is my opinion that this issue is resolved by 2db695b (above)
